<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.56">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Leighton Pritchard">

<title>1&nbsp; Introduction – BM425 Workshop 1: SARS-CoV-2 Genome Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./galaxy-registration.html" rel="next">
<link href="./index.html" rel="prev">
<link href="./icon_32.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./intro.html"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">BM425 Workshop 1: SARS-CoV-2 Genome Analysis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/sipbs-compbiol/BM425-Workshop-1" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface to 2024-25 BM425 Workshop 1</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./galaxy-registration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Register An Account with Galaxy</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./join-session.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Join the Training Session</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./download-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Download Workshop Data</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./upload-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Upload Data to Galaxy</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./late-section.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Late Section</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./webr-example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title"><code>R</code> Playground</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./acknowledgements.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Acknowledgements</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#workshop-scope" id="toc-workshop-scope" class="nav-link active" data-scroll-target="#workshop-scope"><span class="header-section-number">1.1</span> Workshop Scope</a></li>
  <li><a href="#covid-19" id="toc-covid-19" class="nav-link" data-scroll-target="#covid-19"><span class="header-section-number">1.2</span> COVID-19</a></li>
  <li><a href="#sars-cov-2-spike-s-protein" id="toc-sars-cov-2-spike-s-protein" class="nav-link" data-scroll-target="#sars-cov-2-spike-s-protein"><span class="header-section-number">1.3</span> SARS-CoV-2 Spike (S) Protein</a></li>
  <li><a href="#data-provenance" id="toc-data-provenance" class="nav-link" data-scroll-target="#data-provenance"><span class="header-section-number">1.4</span> Data Provenance</a>
  <ul class="collapse">
  <li><a href="#the-first-sequenced-human-covid-19-case" id="toc-the-first-sequenced-human-covid-19-case" class="nav-link" data-scroll-target="#the-first-sequenced-human-covid-19-case"><span class="header-section-number">1.4.1</span> The first sequenced human COVID-19 case</a></li>
  <li><a href="#sars-cov-2-read-data-for-this-workshop" id="toc-sars-cov-2-read-data-for-this-workshop" class="nav-link" data-scroll-target="#sars-cov-2-read-data-for-this-workshop"><span class="header-section-number">1.4.2</span> SARS-CoV-2 read data for this workshop</a></li>
  <li><a href="#sars-cov-2003-genome-data-for-this-workshop" id="toc-sars-cov-2003-genome-data-for-this-workshop" class="nav-link" data-scroll-target="#sars-cov-2003-genome-data-for-this-workshop"><span class="header-section-number">1.4.3</span> SARS-CoV 2003 genome data for this workshop</a></li>
  <li><a href="#march-2020-dutch-sars-cov-2-isolate" id="toc-march-2020-dutch-sars-cov-2-isolate" class="nav-link" data-scroll-target="#march-2020-dutch-sars-cov-2-isolate"><span class="header-section-number">1.4.4</span> March 2020 Dutch SARS-CoV-2 isolate</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/sipbs-compbiol/BM425-Workshop-1/edit/main/intro.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/sipbs-compbiol/BM425-Workshop-1/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="workshop-scope" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="workshop-scope"><span class="header-section-number">1.1</span> Workshop Scope</h2>
<p>Today you will be:</p>
<ul>
<li>assembling, annotating, and visualising the first SARS-CoV-2 genome to be sequenced (from Wuhan, January 2020).</li>
<li>comparing the first SARS-CoV-2 genome to an earlier coronavirus (SARS 2003)</li>
<li>comparing the first SARS-CoV-2 genome and a SARS-CoV-2 genome that was sequenced later in the pandemic</li>
<li>interpreting the biological signifance of differences between the genomes, particularly in the region of the spike (S) protein.</li>
</ul>
</section>
<section id="covid-19" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="covid-19"><span class="header-section-number">1.2</span> COVID-19</h2>
<p>The first cases of a pneumonia of unknown causes began to emerge in Wuhan, China, in December 2019. The novel coronavirus SARS-CoV-2 was isolated by early January, and by the end of January 2020 human-to-human transmission had been confirmed, and cases had been reported in Thailand, Japan, the USA, and several other countries (<a href="#fig-timeline" class="quarto-xref">Figure&nbsp;<span>1.1</span></a>, <span class="citation" data-cites="Wang2020-yy">Wang et al. (<a href="references.html#ref-Wang2020-yy" role="doc-biblioref">2020</a>)</span>).</p>
<div id="fig-timeline" class="quarto-float quarto-figure quarto-figure-center anchored" alt="A timeline from December 30th 2019 to January 24th 2020, detailing the successive identification of a cluster of pneumonia cases, identification of a novel coronavirus, and first cases in Thailand and Korea.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-timeline-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="assets/images/early-covid-timeline.jpg" class="img-fluid figure-img" style="width:80.0%" alt="A timeline from December 30th 2019 to January 24th 2020, detailing the successive identification of a cluster of pneumonia cases, identification of a novel coronavirus, and first cases in Thailand and Korea.">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-timeline-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.1: Early COVID-19 timeline (reproduced from <span class="citation" data-cites="Wang2020-yy">Wang et al. (<a href="references.html#ref-Wang2020-yy" role="doc-biblioref">2020</a>)</span>)
</figcaption>
</figure>
</div>
<p>By the end of February, cases had been identified across the Northern hemisphere, and a new hotspot was emerging in Italy (<a href="#fig-spread" class="quarto-xref">Figure&nbsp;<span>1.2</span></a>).</p>
<div id="fig-spread" class="quarto-float quarto-figure quarto-figure-center anchored" alt="Animated .gif showing the global spread of COVID-19 from China in early January 2020, to most of the Northern hemisphere by the end of February 2020.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-spread-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="assets/images/COVID-19-outbreak-timeline-2.gif" class="img-fluid figure-img" style="width:80.0%" alt="Animated .gif showing the global spread of COVID-19 from China in early January 2020, to most of the Northern hemisphere by the end of February 2020.">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-spread-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.2: Early COVID-19 global spread
</figcaption>
</figure>
</div>
<p>By October 2021, it was estimated that COVID-19 had <a href="https://www.reuters.com/world/global-covid-19-deaths-hit-5-million-delta-variant-sweeps-world-2021-10-02/">caused over 5mn deaths</a> worldwide, as well as severe social and economic disruption, including the <a href="https://www.imf.org/en/Blogs/Articles/2020/04/14/blog-weo-the-great-lockdown-worst-economic-downturn-since-the-great-depression">largest global recession in almost a century</a>.</p>
<p>Despite this huge impact, societal and scientific efforts including lockdowns, mathematical modelling, and novel vaccine development helped avoid even worse outcomes. COVID-19 vaccines are estimated to have prevented an additional 14-20mn deaths worldwide in 2021 (<span class="citation" data-cites="Watson2022-ec">Watson et al. (<a href="references.html#ref-Watson2022-ec" role="doc-biblioref">2022</a>)</span>).</p>
<p>In May 2023, the World Health Organisation (WHO) <a href="https://www.reuters.com/business/healthcare-pharmaceuticals/covid-is-no-longer-global-health-emergency-who-2023-05-05/">ended their Public Health Emergency</a> of International Concern (PHEIC), but the SARS-CoV-2 virus remains in circulation. As we begin this workshop, SARS-CoV-2 has <a href="https://ourworldindata.org/coronavirus">claimed over 7mn lives</a> (<span class="citation" data-cites="owidcoronavirus">Mathieu et al. (<a href="references.html#ref-owidcoronavirus" role="doc-biblioref">2020</a>)</span>).</p>
</section>
<section id="sars-cov-2-spike-s-protein" class="level2" data-number="1.3">
<h2 data-number="1.3" class="anchored" data-anchor-id="sars-cov-2-spike-s-protein"><span class="header-section-number">1.3</span> SARS-CoV-2 Spike (S) Protein</h2>
<p>The spike (S) protein of SARS-CoV-2 plays a central role in host cell receptor recognition and cell membrane fusion (<span class="citation" data-cites="Huang2020-uk">Huang et al. (<a href="references.html#ref-Huang2020-uk" role="doc-biblioref">2020</a>)</span>). The protein comprises two subunits:</p>
<ul>
<li>S1: a receptor-binding domain that recognises and binds ACE2 (angiotensin-converting enzyme 2)</li>
<li>S2: a domain that mediates membrane fusion between the virus and the cell by forming a structural bundle</li>
</ul>
<p>which interact with the host cell as illustrated in <a href="#fig-spike" class="quarto-xref">Figure&nbsp;<span>1.3</span></a>.</p>
<div id="fig-spike" class="quarto-float quarto-figure quarto-figure-center anchored" alt="Schematic of SARS-CoV-2 interaction with the host cell, mediated by the spike (S) protein. The S protein attaches to the ACE receptor _via_ the S1 subunit, and the S2 subunit then forms a structural bundle, interacting with the host cell to permit virus entry.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-spike-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="assets/images/spike_protein_cell.png" class="img-fluid figure-img" style="width:80.0%" alt="Schematic of SARS-CoV-2 interaction with the host cell, mediated by the spike (S) protein. The S protein attaches to the ACE receptor _via_ the S1 subunit, and the S2 subunit then forms a structural bundle, interacting with the host cell to permit virus entry.">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-spike-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.3: (a) The schematic structure of the S protein. (b) The S protein binds to the receptor ACE2. (c) The binding and virus–cell fusion process mediated by the S protein. (d) The life cycle of SARS-CoV-2 in host cells (reproduced from <span class="citation" data-cites="Huang2020-uk">Huang et al. (<a href="references.html#ref-Huang2020-uk" role="doc-biblioref">2020</a>)</span>)
</figcaption>
</figure>
</div>
<p>When you assemble, annotate, and visualise the SARS-CoV-2 genome in this workshop you will find the spike protein annotated as in <a href="#fig-jbrowse-1" class="quarto-xref">Figure&nbsp;<span>1.4</span></a>.</p>
<div id="fig-jbrowse-1" class="quarto-float quarto-figure quarto-figure-center anchored" alt="JBrowse view of an annotated SARS-CoV-2 assembly, showing the location of the spike (S) protein on the genome.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-jbrowse-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="assets/images/spike_vis_jbrowse.png" class="img-fluid figure-img" alt="JBrowse view of an annotated SARS-CoV-2 assembly, showing the location of the spike (S) protein on the genome.">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-jbrowse-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.4: JBrowse rendering of an annotated SARS-CoV-2 genome
</figcaption>
</figure>
</div>
<p>The spike protein is not the only important surface protein for SARS-CoV-2. The envelope (E), membrane (M) and nucleocapsid proteins are also essential for the virus to interact with the host, as shown in <a href="#fig-palma" class="quarto-xref">Figure&nbsp;<span>1.5</span></a>, and you will identify these in your analysis today, too.</p>
<div id="fig-palma" class="quarto-float quarto-figure quarto-figure-center anchored" alt="Schematic representation of SARS-CoV-2 surface proteins and their interactions with the host.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-palma-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="assets/images/sars-ace2tmprs2-martapalma.png" class="img-fluid figure-img" style="width:80.0%" alt="Schematic representation of SARS-CoV-2 surface proteins and their interactions with the host.">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-palma-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.5: How SARS-CoV-2 interacts with the surface proteins of the target cell <a href="https://pdb101.rcsb.org/news/5ed7c17cdab5c9354c274f56">(Marta Palma Rodríguez)</a>
</figcaption>
</figure>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Other SARS-CoV-2 Structures
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>You can find an interactive collection of publicly-available SARS-CoV-2 protein structures at <a href="https://www.rcsb.org/">RCSB/PDB</a>, the public repository for biological structural data.</p>
<ul>
<li><a href="https://www.rcsb.org/news/feature/5e74d55d2d410731e9944f52">SARS-CoV-2 PDB Structures</a></li>
</ul>
</div>
</div>
</div>
</section>
<section id="data-provenance" class="level2" data-number="1.4">
<h2 data-number="1.4" class="anchored" data-anchor-id="data-provenance"><span class="header-section-number">1.4</span> Data Provenance</h2>
<p>Modern scientific research, and this workshop, are both made possible due to the dedication of scientists around the globe to <a href="https://www.go-fair.org/fair-principles/">FAIR (Findable, Accessible, Interoperable, and Reproducible) principles for scientific data management</a>.</p>
<p>Throughout the COVID-19 pandemic, SARS-CoV-2 sequence and structural data was released through public repositories including <a href="https://www.ncbi.nlm.nih.gov/">NCBI</a>, <a href="https://gisaid.org/">GISAID</a>, and <a href="https://www.rcsb.org/">RCSB/PDB</a>. The data shared through these resources enabled scientists across the world to conduct research that reduced suffering and hastened the end of the pandemic. These practices were established before COVID-19, and continue to support and empower scientific research globally.</p>
<section id="the-first-sequenced-human-covid-19-case" class="level3" data-number="1.4.1">
<h3 data-number="1.4.1" class="anchored" data-anchor-id="the-first-sequenced-human-covid-19-case"><span class="header-section-number">1.4.1</span> The first sequenced human COVID-19 case</h3>
<p>The first SARS-CoV-2 genome was obtained from a sample of human lung in January 2020. The sample was sequenced and submitted to NCBI as <a href="https://www.ncbi.nlm.nih.gov/biosample/?term=SAMN13922059">SAMN13922059</a>. It is recorded as “human lung metagenome” because the entire material - human, virus, and everything else - was sequenced as a single sample. The work as described in <span class="citation" data-cites="Wu2020-yx">Wu et al. (<a href="references.html#ref-Wu2020-yx" role="doc-biblioref">2020</a>)</span> is outlined below.</p>
<ol type="1">
<li>The lung was lavaged (washed/rinsed) and the fluid collected</li>
<li>Total RNA was extracted from the fluid, and an RNA sequencing library constructed (including depletion of ribosomal RNA)</li>
<li>The library was sequenced on an <a href="https://www.illumina.com/systems/sequencing-platforms/miniseq.html">Illumina MiniSeq</a> machine, to give 150bp paired-end reads</li>
<li>The resulting reads were cleaned and assembled using Megahit (<span class="citation" data-cites="Li2015-yc">Li et al. (<a href="references.html#ref-Li2015-yc" role="doc-biblioref">2015</a>)</span>) and <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki">Trinity</a></li>
<li>The longest contig was identified as a coronavirus, and used to design primers for PCR-RACE (Rapid Amplification of cDNA Ends, <span class="citation" data-cites="Pal2022-tw">Pal (<a href="references.html#ref-Pal2022-tw" role="doc-biblioref">2022</a>)</span>, <a href="#fig-race" class="quarto-xref">Figure&nbsp;<span>1.6</span></a>) to sequence the ends of the virus genome</li>
<li>The cleaned reads were remapped against the full virus sequence</li>
</ol>
<div id="fig-race" class="quarto-float quarto-figure quarto-figure-center anchored" alt="Schematic of how PCR-RACE leads to a crenellated assembly, with long reads spanning the virus genome.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-race-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="assets/images/race.png" class="img-fluid figure-img" style="width:80.0%" alt="Schematic of how PCR-RACE leads to a crenellated assembly, with long reads spanning the virus genome.">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-race-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.6: PCR-RACE generates long amplicons from the viral genome, which can be sequenced with long-read sequencing. This results in a “crenellated” assembly of overlapping long genomic sections.
</figcaption>
</figure>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Questions
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="1">
<li>Why did they extract RNA and not DNA?</li>
<li>Why do you think they depleted the ribosomal RNA?</li>
</ol>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Answers
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<ol type="1">
<li>SARS-CoV-2 is an RNA virus, so we must sequence RNA to recover virus sequence.</li>
<li>There is so much ribosomal RNA in biological material that, if it was left in the sample, it would dominate over the viral material and any human transcripts, making the virus more difficult to detect. Depleting it enhances sensitivity for detecting viral RNA.</li>
</ol>
</div>
</div>
</div>
</section>
<section id="sars-cov-2-read-data-for-this-workshop" class="level3" data-number="1.4.2">
<h3 data-number="1.4.2" class="anchored" data-anchor-id="sars-cov-2-read-data-for-this-workshop"><span class="header-section-number">1.4.2</span> SARS-CoV-2 read data for this workshop</h3>
<p>For this workshop, we took all 57mn raw sequencing reads from <span class="citation" data-cites="Wu2020-yx">Wu et al. (<a href="references.html#ref-Wu2020-yx" role="doc-biblioref">2020</a>)</span>, which are available at NCBI under accession <a href="https://www.ncbi.nlm.nih.gov/sra/?term=SRR10971381">SRR10971381</a>, and filtered out all reads that matched the human genome. The remaining reads in the files below are those you will use to assemble the SARS-CoV-2 genome.</p>
<ul>
<li><a href="assets/data/trimmed_pe_aln.qsorted.mapped.fixed.1.fastq.gz">Paired-end reads, forward direction</a></li>
<li><a href="assets/data/trimmed_pe_aln.qsorted.mapped.fixed.2.fastq.gz">Paired-end reads, reverse direction</a></li>
</ul>
</section>
<section id="sars-cov-2003-genome-data-for-this-workshop" class="level3" data-number="1.4.3">
<h3 data-number="1.4.3" class="anchored" data-anchor-id="sars-cov-2003-genome-data-for-this-workshop"><span class="header-section-number">1.4.3</span> SARS-CoV 2003 genome data for this workshop</h3>
<p>The first comparator genome in the workshop is that of SARS coronavirus Urbani. This coronavirus emerged in Guangdong, China, in late 2002 and was the subject of a World Health Organisation global alert, which named it SARS (Severe Acute Respiratory Syndrome). This strain of SARS-CoV was sequenced in 2003 (<span class="citation" data-cites="Rota2003-tu">Rota et al. (<a href="references.html#ref-Rota2003-tu" role="doc-biblioref">2003</a>)</span>) and has accession <a href="https://www.ncbi.nlm.nih.gov/nuccore/AY278741">AY278741.1</a> at GenBank/NCBI.</p>
<p>In the workshop, you will use the following files:</p>
<ul>
<li><a href="assets/data/SARS_AY278741.fasta">SARS 2003 genome sequence (<code>.fasta</code>)</a></li>
<li><a href="assets/data/SARS_AY278741.gff3">SARS 2003 annotation (<code>.gff3</code>)</a></li>
<li><a href="assets/data/SARS_AY278741.gbk">SARS 2003 combined genome sequence and annotation (<code>.gbk</code>)</a></li>
</ul>
</section>
<section id="march-2020-dutch-sars-cov-2-isolate" class="level3" data-number="1.4.4">
<h3 data-number="1.4.4" class="anchored" data-anchor-id="march-2020-dutch-sars-cov-2-isolate"><span class="header-section-number">1.4.4</span> March 2020 Dutch SARS-CoV-2 isolate</h3>
<p>The second comparator genome you will use in this workshop is a SARS-CoV-2 variant sequenced in the Netherlands in March 2020, four months after the first known human cases were identified. The sample has accession <a href="https://www.ncbi.nlm.nih.gov/biosample/SAMEA6847958">SAMEA6847958</a> and was obtained through active surveillance during the COVID-19 pandemic (<span class="citation" data-cites="Oude_Munnink2020-ry">Oude Munnink et al. (<a href="references.html#ref-Oude_Munnink2020-ry" role="doc-biblioref">2020</a>)</span>).</p>
<p>The data describes a set of <a href="https://nanoporetech.com/products/sequence/gridion">Oxford Nanopore (ONT) GridION</a> reads, which are available under accession <a href="https://trace.ncbi.nlm.nih.gov/Traces/?view=run_browser&amp;acc=ERR4164769&amp;display=metadata">ERR4164769</a> at SRA/NCBI. In the workshop, you will use the following file:</p>
<ul>
<li><a href="assets/data/ERR4164769_1.fastq.gz">GridION reads for March 2020 Dutch isolate</a></li>
</ul>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-Huang2020-uk" class="csl-entry" role="listitem">
Huang, Yuan, Chan Yang, Xin-Feng Xu, Wei Xu, and Shu-Wen Liu. 2020. <span>“Structural and Functional Properties of <span>SARS-CoV-2</span> Spike Protein: Potential Antivirus Drug Development for <span>COVID-19</span>.”</span> <em>Acta Pharmacol. Sin.</em> 41 (9): 1141–49.
</div>
<div id="ref-Li2015-yc" class="csl-entry" role="listitem">
Li, Dinghua, Chi-Man Liu, Ruibang Luo, Kunihiko Sadakane, and Tak-Wah Lam. 2015. <span>“<span>MEGAHIT</span>: An Ultra-Fast Single-Node Solution for Large and Complex Metagenomics Assembly via Succinct de Bruijn Graph.”</span> <em>Bioinformatics</em> 31 (10): 1674–76.
</div>
<div id="ref-owidcoronavirus" class="csl-entry" role="listitem">
Mathieu, Edouard, Hannah Ritchie, Lucas Rodés-Guirao, Cameron Appel, Charlie Giattino, Joe Hasell, Bobbie Macdonald, et al. 2020. <span>“Coronavirus Pandemic (COVID-19).”</span> <em>Our World in Data</em>.
</div>
<div id="ref-Oude_Munnink2020-ry" class="csl-entry" role="listitem">
Oude Munnink, Bas B, David F Nieuwenhuijse, Mart Stein, Áine O’Toole, Manon Haverkate, Madelief Mollers, Sandra K Kamga, et al. 2020. <span>“Rapid <span>SARS-CoV-2</span> Whole-Genome Sequencing and Analysis for Informed Public Health Decision-Making in the Netherlands.”</span> <em>Nat. Med.</em> 26 (9): 1405–10.
</div>
<div id="ref-Pal2022-tw" class="csl-entry" role="listitem">
Pal, Aruna. 2022. <span>“Rapid Amplification of <span class="nocase">cDNA</span> Ends (<span>RACE</span>).”</span> In <em>Springer Protocols Handbooks</em>, 505–35. New York, NY: Springer US.
</div>
<div id="ref-Rota2003-tu" class="csl-entry" role="listitem">
Rota, Paul A, M Steven Oberste, Stephan S Monroe, W Allan Nix, Ray Campagnoli, Joseph P Icenogle, Silvia Peñaranda, et al. 2003. <span>“Characterization of a Novel Coronavirus Associated with Severe Acute Respiratory Syndrome.”</span> <em>Science</em> 300 (5624): 1394–99.
</div>
<div id="ref-Wang2020-yy" class="csl-entry" role="listitem">
Wang, Chen, Peter W Horby, Frederick G Hayden, and George F Gao. 2020. <span>“A Novel Coronavirus Outbreak of Global Health Concern.”</span> <em>Lancet</em> 395 (10223): 470–73.
</div>
<div id="ref-Watson2022-ec" class="csl-entry" role="listitem">
Watson, Oliver J, Gregory Barnsley, Jaspreet Toor, Alexandra B Hogan, Peter Winskill, and Azra C Ghani. 2022. <span>“Global Impact of the First Year of <span>COVID-19</span> Vaccination: A Mathematical Modelling Study.”</span> <em>Lancet Infect. Dis.</em> 22 (9): 1293–1302.
</div>
<div id="ref-Wu2020-yx" class="csl-entry" role="listitem">
Wu, Fan, Su Zhao, Bin Yu, Yan-Mei Chen, Wen Wang, Zhi-Gang Song, Yi Hu, et al. 2020. <span>“A New Coronavirus Associated with Human Respiratory Disease in China.”</span> <em>Nature</em> 579 (7798): 265–69.
</div>
</div>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./index.html" class="pagination-link" aria-label="Preface to {{< var ay >}} BM425 Workshop 1">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Preface to 2024-25 BM425 Workshop 1</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./galaxy-registration.html" class="pagination-link" aria-label="Register An Account with Galaxy">
        <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Register An Account with Galaxy</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>This workshop was written by Dr Leighton Pritchard, Dr Morgan Feeney, and Dr Nick Tucker</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/sipbs-compbiol/BM425-Workshop-1/edit/main/intro.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/sipbs-compbiol/BM425-Workshop-1/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This book was built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




</body></html>